lib/gollum-lib/macro/video.rb

```diff
@@ -0,0 +1,17 @@
+module Gollum
+  class Macro
+    class Video < Gollum::Macro
+      def render (fname)
+        begin 
```

Comment:
The code does both, checks if the video file exists and also detects the video's MIME type so users don't need to specify it explicitly in the macro. With the current version of the macro, it generates the correct video/#{type} based on the video file being linked to.  We could detect the type based on a file extension, but that won't be considered a reliable way of detecting file types. This is also the reason why I committed video test files to the repo; a test case to cover the automatic mime type detection part. Regarding a bare repo and when a file doesn't exist in the file system, we could add a `File.exists?(pname)` before trying to open, but then what should we use for the mime type?  I am open to any suggestions. Thanks